src/Microsoft.OData.Core/TypeUtils.cs

```diff
@@ -96,7 +96,7 @@ internal static bool AreTypesEquivalent(Type typeA, Type typeB)
         /// <param name="isCollection">Returns whether or not the returned type is a collection.</param>
         internal static void ParseQualifiedTypeName(string qualifiedTypeName, out string namespaceName, out string typeName, out bool isCollection)
         {
-            isCollection = qualifiedTypeName.StartsWith(ODataConstants.CollectionPrefix + "(", StringComparison.Ordinal);
+            isCollection = qualifiedTypeName.StartsWith(string.Concat(ODataConstants.CollectionPrefix, "("), StringComparison.Ordinal);

```

Comment:
>string.Concat(ODataConstants.CollectionPrefix, "(") [](start = 56, length = 51)

If this code is called frequently, since all of the strings being concatenated are static, a better optimization would be to define an ODataConstantsCollectionPrefixWithParens = string.Concat(ODataContstants.CollectionPrefix, "(")); and use that here (so the concatenation only occurs once).

Even better, in this case, it looks like in every case that ODataConstants.CollectionPrefix is used, the parens is added (except when the length is used, in which case the count is incremented by one to account for the open parens).  So you could just define the ODataConstants.CollectionPrefix to include the parens and then fix the places it is referenced to account or the fact that the open parens is already added.